modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -228,15 +230,20 @@ public static void checkInstantiable(Class<?> declaringClass)
 
     public static void checkInstantiable(Class<?> declaringClass, boolean requireDefaultConstructor)
     {
-        checkArgument(declaringClass != null, "declaringClass cannot be null");
+        checkArgument(isInstantiable(declaringClass, requireDefaultConstructor), String.format("Class %s cannot be instantiated.", declaringClass));
```

Comment:
replace with an if. If you do it like this, you pay the price of the string.format even if the validation passes